From 31e9e1746755852a83eda500f705cbc7d78ac468 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Thu, 28 Jun 2018 16:40:57 +0200
Subject: [PATCH] mtd: spi-nor: Add Winbond w25q128jv support

Datasheet:
http://www.winbond.com/resource-files/w25q128jv%20revf%2003272018%20plus.pdf

Testing done on Mikrotik Routerboard  wAP R board.
It does not support Dual or Quad modes.
---
 ...td-spi-nor-Add-support-for-w25q128jv.patch | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 target/linux/generic/backport-4.9/062-pending-0009-mtd-spi-nor-Add-support-for-w25q128jv.patch

diff --git a/target/linux/generic/backport-4.9/062-pending-0009-mtd-spi-nor-Add-support-for-w25q128jv.patch b/target/linux/generic/backport-4.9/062-pending-0009-mtd-spi-nor-Add-support-for-w25q128jv.patch
new file mode 100644
index 00000000000..fd70e61efd1
--- /dev/null
+++ b/target/linux/generic/backport-4.9/062-pending-0009-mtd-spi-nor-Add-support-for-w25q128jv.patch
@@ -0,0 +1,34 @@
+From b3b0dd8da363308355c0d7fcd3270962c05bd6d2 Mon Sep 17 00:00:00 2001
+From: Robert Marko <robimarko@gmail.com>
+Date: Sat, 23 Jun 2018 12:08:27 +0200
+Subject: [PATCH] mtd: spi-nor: Add Winbond w25q128jv support
+
+mtd: spi-nor: Add Winbond w25q32jv support
+
+Datasheet:
+http://www.winbond.com/resource-files/w25q128jv%20revf%2003272018%20plus.pdf
+
+Testing done on Mikrotik Routerboard  wAP R board.
+It does not support Dual or Quad modes.
+
+Signed-off-by: Robert Marko <robimarko@gmail.com>
+---
+ spi-nor.c | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
+index 9a6a6a8..d4983e0 100644
+--- a/drivers/mtd/spi-nor/spi-nor.c
++++ b/drivers/mtd/spi-nor/spi-nor.c
+@@ -1149,6 +1149,11 @@ static const struct flash_info spi_nor_ids[] = {
+ 			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+ 			SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB)
+ 	},
++	{ 
++		"w25q128jv", INFO(0xef7018, 0, 64 * 1024, 256,
++			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
++			SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB)
++	},
+ 	{ "w25q80", INFO(0xef5014, 0, 64 * 1024,  16, SECT_4K) },
+ 	{ "w25q80bl", INFO(0xef4014, 0, 64 * 1024,  16, SECT_4K) },
+ 	{ "w25q128", INFO(0xef4018, 0, 64 * 1024, 256, SECT_4K) },
